subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -75,12 +75,20 @@ static bool parse_number(const struct shell *sh, long *param, char *str, long mi
 {
 	char *endptr;
 	char *str_tmp = str;
-	long num = strtol(str_tmp, &endptr, 10);
+	long num = 0;
+
+	if ((str_tmp[0] == '0') && (str_tmp[1] == 'x')) {
+		/* Hexadecimal numbers take base 0 in strtol */
+		num = strtol(str_tmp, &endptr, 0);
+	} else {
+		num = strtol(str_tmp, &endptr, 10);
+	}
```

Comment:
According to `strtol()` documentation, the function tries to guess what the format of the number is if 0 base is used. If the string starts with 0x it is assumed hex, if it starts with 0 it is assumed octal, and in the other cases it is base 10. So at least in theory we should be able to use `num = strtol(str_tmp, &endptr, 0);` always, could you try if that works ok for both hex and base 10 values.

Reply:
will check this

Reply:
Just passing Zero as base works for both base 10 and hex. Making that change

Reply:
During CI, It was found that using base 0 as the strtol base fails for certain builds. placing a hexadecimal check so that the function will be able to address hexadecimal parameters going forward